qiime2/plugin/plugin.py

```diff
@@ -79,7 +83,7 @@ def __init__(self, name, version, website, package=None, project_name=None,
         self.views = {}
         self.type_fragments = {}
         self.transformers = {}
-        self.type_formats = []
+        self.artifact_classes = []
```

Comment:
This might be better as a dictionary for some of the registration behavior

Reply:
Good call - thanks for catching that! Made this change and most of my commit responding to your comments is updates to address this.